pandas/io/parsers.py

```diff
@@ -1176,13 +1176,18 @@ def _should_parse_dates(self, i):
         if isinstance(self.parse_dates, bool):
             return self.parse_dates
         else:
-            name = self.index_names[i]
+            if self.index_names is not None:
+                name = self.index_names[i]
```

Comment:
`index_names` is `None` by default, though why that wasn't caught before is strange.